### PR TITLE
tiles: moving common components to @loaders.gl/tiles module [Part 1]

### PR DESCRIPTION
**Pull request by xintongxia**
For #631 

`@loaders.gl/tiles` is the tiles category module

- move common components, `TileTraversal`, `TileCache` and other utility functions and classes to `@loaders.gl/tiles`

Next steps:
- Consolidate `Tileset`, and `Tile` classes and move to `@loaders.gl/tiles`

### CODE REVIEW

File: modules/3d-tiles/package.json [hunk 1]
```diff
@@ -36,14 +36,12 @@
...
     "@loaders.gl/loader-utils": "2.1.0-alpha.3",
     "@loaders.gl/math": "2.1.0-alpha.3",
     "@math.gl/culling": "^3.1.2",
     "@math.gl/geospatial": "^3.1.2",
-    "math.gl": "^3.1.2",
-    "probe.gl": "^3.2.0"
-  },
-  "devDependencies": {
-    "@deck.gl/core": "^8.0.0"
+    "@math.gl/core": "^3.1.3",
```
Comment (ibgreen) [inline modules/3d-tiles/package.json]: Why dev dependencies?

File: modules/3d-tiles/src/lib/parsers/helpers/normalize-3d-tile-normals.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import {Vector3} from 'math.gl';
+import {Vector3} from '@math.gl/core';
```
Comment (ibgreen) [inline modules/3d-tiles/src/lib/parsers/helpers/normalize-3d-tile-normals.js]: Not a devDependency...

File: modules/3d-tiles/src/lib/tileset/tile-3d-header.js [hunk 3]
```diff
@@ -1,14 +1,16 @@
 // This file is derived from the Cesium code base under Apache 2 license
 // See LICENSE.md and https://github.com/AnalyticalGraphicsInc/cesium/blob/master/LICENSE.md
-// import {TILE3D_REFINEMENT, TILE3D_OPTIMIZATION_HINT} from '../constants';
-import {Vector3, Matrix4} from 'math.gl';
+// import {TILE_REFINEMENT, TILE3D_OPTIMIZATION_HINT} from '../constants';
```
Comment (ibgreen) [inline modules/3d-tiles/src/lib/tileset/tile-3d-header.js]: Cur?

File: modules/3d-tiles/src/lib/tileset/tileset-3d.js [hunk 4]
```diff
@@ -1,19 +1,15 @@
 // This file is derived from the Cesium code base under Apache 2 license
 // See LICENSE.md and https://github.com/AnalyticalGraphicsInc/cesium/blob/master/LICENSE.md
 
-import {Matrix4, Vector3} from 'math.gl';
+import {Matrix4, Vector3} from '@math.gl/core';
 import {Ellipsoid} from '@math.gl/geospatial';
-import {Stats} from 'probe.gl';
+import {Stats} from '@probe.gl/stats';
```
Comment (ibgreen) [inline modules/3d-tiles/src/lib/tileset/tileset-3d.js]: Aren't you moving `Tileset3D` to the `tiles` module per your RFC? Or that is "phase 2"?

File: modules/i3s/src/index.js [hunk 5]
```diff
@@ -1,5 +1,3 @@
-export {I3SNodeGeometryLoader} from './i3s-node-geometry-loader';
```
Comment (ibgreen) [inline modules/i3s/src/index.js]: What happened to these?

File: modules/tiles/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Overview
```
Comment (ibgreen) [inline modules/tiles/docs/README.md]: You have excellent material in your RFC that you can just copy in here, plus move/copy some of the docs from 3d-tiles...

File: modules/3d-tiles/package.json (hunk 1 above)
Comment (xintongxia) [reply modules/3d-tiles/package.json]: there is no `devDependencies` in `3d-tiles` module

File: modules/3d-tiles/src/lib/parsers/helpers/normalize-3d-tile-normals.js (hunk 2 above)
Comment (xintongxia) [reply modules/3d-tiles/src/lib/parsers/helpers/normalize-3d-tile-normals.js]: yes, it is a `dependency`

File: modules/3d-tiles/src/lib/tileset/tileset-3d.js (hunk 4 above)
Comment (xintongxia) [reply modules/3d-tiles/src/lib/tileset/tileset-3d.js]: yes, it will be in the following PR

File: modules/i3s/src/index.js (hunk 5 above)
Comment (xintongxia) [reply modules/i3s/src/index.js]: the `I3SNodeGeometryLoader` is not implemented yet, so just cut it for now.

**Comment by xintongxia** [comment]
> * (`@loaders.gl/image`) and keep `@loaders.gl/images` as the category module for image loaders, with the extra API.

Separate out to category modules, should this go to v2.1 or v3.0?

**Comment by ibgreen** [comment]
> Separate <images> out to category modules, should this go to v2.1 or v3.0?

If agree to do this, it would be a breaking change, since the ImageLoader would be moved out of the `@loaders.gl/images` module.

But if we re-exported the ImageLoader in the images module (for now), then there would be no breaking change, and it could go into v2.x.

File: modules/tiles/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    - `maximumMemoryUsage`=`512`] (`Number`) - The maximum amount of memory in MB that can be used by the tileset.
+    - `ellipsoid`=`Ellipsoid.WGS84` ([`Ellipsoid`](https://math.gl/modules/geospatial/docs/api-reference/ellipsoid)) - The ellipsoid determining the size and shape of the globe.
+      Callbacks:
+    - `onTileLoad` (`(tileHeader : Tile3DHeader) : void`) - callback when a tile node is fully loaded during the tileset traversal.
+    - `onTileUnload` (`(tileHeader : Tile3DHeader) : void`) - callback when a tile node is unloaded during the tileset traversal.
+    - `onTileError` (`(tileHeader : Tile3DHeader, message : String) : void`) - callback when a tile faile to load during the tileset traversal.
+- `update(viewport: WebMercatorViewport) : Number`: Execute traversal under current viewport and fetch tiles needed for current viewport and update `selectedTiles`. Return `frameNumber` of this update frame.
+- `destroy() : void`: Destroys the WebGL resources held by this object, and destroy all the tiles' resources by recursively traversing the tileset tree.
+
+### Tile3D Class
```
Comment (Pessimistress) [inline modules/tiles/docs/README.md]: The documentation for each class should be split out to their own files.

Comment (xintongxia) [reply modules/tiles/docs/README.md]: will do it in the following PR.

**Pull request merged**